### PR TITLE
add asset description to asset objects

### PR DESCRIPTION
**Pull request by matthewhanson**
**Related Issue(s):** #
#663 

**Proposed Changes:**

Assets have a title, which is very useful for displaying it's name in a UI. However a description can provide additional important info about the asset

1. Adds a description to assets. 
2. Also adds `description` and `role` to the Asset definition (so it aligns with an Item Asset except for `href`)

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: item-spec/item-spec.md [hunk 1]
```diff
@@ -168,12 +168,14 @@ or streamed. It is allowed to add additional fields.
...
-| type       | string | [Media type](#media-types) of the asset.                                              |
-| role        | string | The semantic purpose of the asset.                                                   |
+| title      | string | The displayed title for clients and users                                           |
+| description      | string | A description of the Asset                                            |
+| type       | string | [Media type](#media-types) of the asset                                             |
+| role        | string | The semantic purpose of the asset                                                  |
 
-**role** is intended to be a value that describes semantic usage of an asset, similarly to the use of `rel` in 
-Link. 
+**description** is more verbose than a simple title and could provide details on the asset, such as how it was processed
```
Comment (m-mohr) [inline item-spec/item-spec.md]: These two descriptions are pretty short and could be added to the table for simplicity, right?

Comment (matthewhanson) [reply item-spec/item-spec.md]: +1

**Comment by matthewhanson** [comment]
@m-mohr added changes

**Comment by anayeaye** [comment]
I think also `role` should be added to Asset Definition Extension https://github.com/radiantearth/stac-spec/issues/661#issuecomment-554529690 but that could be a separate PR

**Comment by matthewhanson** [comment]
@m-mohr @anayeaye Added description and role to the Asset definition

**Pull request merged**